graylog2-server/src/main/java/org/graylog2/inputs/InputServiceImpl.java

```diff
@@ -142,6 +142,9 @@ public Input create(Map<String, Object> fields) {
 
     @Override
     public Input find(String id) throws NotFoundException {
+        if (!ObjectId.isValid(id)) {
+            throw new NotFoundException("Input id <" + id + "> is invalid!");
```

Comment:
I agree to the second point, but not to the first. From a separation of responsibility, `InputsResource` shouldn't bother with the id format, but if the entity is present or not. Therefore the way the code is currently structured throwing this exception is perfectly fine. Having a proper deserializer for `ObjectId` later on - sure. But not in the scope of this fix.